model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanSingleUseObjectProviderFactory.java

```diff
@@ -86,7 +71,15 @@ public void init(Config.Scope config) {
 
     @Override
     public void postInit(KeycloakSessionFactory factory) {
-
+        // It is necessary to put the cache initialization here, otherwise the cache would be initialized lazily, that
+        // means also listeners will start only after first cache initialization - that would be too late
+        if (singleUseObjectCache == null) {
+            synchronized (this) {
+                if (singleUseObjectCache == null) {
```

Comment:
This synchronization can be removed since `postInit` is executed only once at the initialization in a single thread, so contrary to lazy init, there is only a single caller.

Reply:
Removed